lib/jobs/wait-sel-events-job.js

```diff
@@ -0,0 +1,213 @@
+// Copyright 2017, Dell EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = waitSelEventJobFactory;
+di.annotate(waitSelEventJobFactory, new di.Provide('Job.Wait.Sel.Events'));
+di.annotate(waitSelEventJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_',
+    'Services.Waterline'
+));
+
+function waitSelEventJobFactory(
+    BaseJob,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _,
+    waterline
+) {
+    var logger = Logger.initialize(waitSelEventJobFactory);
+
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function waitSelEventJob(options, context, taskId) {
+        waitSelEventJob.super_.call(this, logger, options, context, taskId);
+        assert.arrayOfObject(options.alertFilters, 'alertFilters should an array of object');
+        this.options = options;
+        this.nodeId = context.target;
+        this.pollerId = undefined;
+        this.alertFilters = options.alertFilters;
+        this.pollInterval = options.pollInterval || 60000;
+        this.legacyInterval = undefined;
+        this.freshAlerts = [];
+    }
+    util.inherits(waitSelEventJob, BaseJob);
+
+    /**
+     * @memberOf waitSelEventJob
+     */
+    waitSelEventJob.prototype._run = function run() {
+        var self = this;
+        return waterline.workitems.findPollers({
+            node: self.nodeId,
+            'config.command': 'selEntries'
+        })
+        .tap(function(pollers){
+            self._retrievePollerInfo(pollers);
```

Comment:
this promise isn't returned! Firstly the timing is a problem, then if this promise is rejected, the `catch` cannot catch the error and process will crash.